src/core/load_balancing/pick_first/pick_first.cc

```diff
@@ -1067,6 +1068,7 @@ void PickFirst::SubchannelList::StartConnectingNextSubchannel() {
       sc->RequestConnectionWithTimer();
       return;
     }
+    sc->set_seen_transient_failure();
```

Comment:
Is this `set_seen_transient_failure` strictly necessary?

Without this, as far as I can think, channels initially in TRANSIENT_FAILURE would switch to IDLE sooner or later at which point we'd proceed with happy eyeballs?

Reply:
Quoting [gRFC A61](https://github.com/grpc/proposal/blob/master/A61-IPv4-IPv6-dualstack-backends.md#happy-eyeballs-in-the-pick_first-lb-policy):

"""
Therefore, when we are ready to start a connection attempt on a given subchannel:
[...]
- If the subchannel is in state TRANSIENT_FAILURE, then we know that it is in backoff due to a recent connection attempt failure, so we treat it as if we have already made a connection attempt on this subchannel, and we will immediately move on to the next subchannel.

"""